arch/risc-v/src/common/riscv_swint.c

```diff
@@ -144,7 +148,11 @@ static void dispatch_syscall(void)
      " addi sp, sp, 4\n"          /* Destroy the stack frame */
      " mv   a2, a0\n"             /* a2=Save return value in a0 */
      " li   a0, 3\n"              /* a0=SYS_syscall_return (3) */
+#ifdef CONFIG_ARCH_USE_S_MODE
```

Comment:
look like we can merge 32bit and 64bit implementation into one by always subtracting 16B.

Reply:
I considered this already and it's not that easy, note that the store/load instructions are also different:

" sd ra, 0(sp)\n" /* Save ra in the stack frame */
vs 
" sw ra, 0(sp)\n" /* Save ra in the stack frame */

There are macros for those, but embedding the macros into this code is not simple.

Also, this is how the implementation was so I did not modify it too much.

Reply:
Ok, let's try to do it in new patch.